minio.spec

```diff
@@ -1,83 +1,64 @@
 URL:            https://www.min.io/
-BuildRequires:  golang >= 1.7
+Requires(pre):  /usr/sbin/useradd, /usr/bin/getent
+Requires(postun): /usr/sbin/userdel
 BuildRoot:      %{tmpdir}/%{name}-%{version}-root-%(id -u -n)
 
 ## Disable debug packages.
 %define         debug_package %{nil}
 
-## Go related tags.
-%define         gobuild(o:) go build -ldflags "${LDFLAGS:-}" %{?**};
-%define         gopath          %{_libdir}/golang
-%define         import_path     github.com/minio/minio
-
 %description
-MinIO is an object storage server released under Apache License v2.0.
-It is compatible with Amazon S3 cloud storage service. It is best
-suited for storing unstructured data such as photos, videos, log
-files, backups and container / VM images. Size of an object can
-range from a few KBs to a maximum of 5TiB.
-
-%prep
-%setup -qc
-mv %{name}-*/* .
-
-install -d src/$(dirname %{import_path})
-ln -s ../../.. src/%{import_path}
-
-%build
-export GOPATH=$(pwd)
+MinIO is a High Performance Object Storage released under Apache License v2.0.
+It is API compatible with Amazon S3 cloud storage service. Use MinIO to build
+high performance infrastructure for machine learning, analytics and application
+data workloads.
 
-# setup flags like 'go run buildscripts/gen-ldflags.go' would do
-tag=%{tag}
-version=${tag#RELEASE.}
-commitid=%{commitid}
-scommitid=$(echo $commitid | cut -c1-12)
-prefix=%{import_path}/cmd
+%pre
+/usr/bin/getent group minio-user || /usr/sbin/groupadd -r minio-user
+/usr/bin/getent passwd minio-user || /usr/sbin/useradd -r -d /etc/minio -s /sbin/nologin minio-user
 
-LDFLAGS="
--X $prefix.Version=$version
--X $prefix.ReleaseTag=$tag
--X $prefix.CommitID=$commitid
--X $prefix.ShortCommitID=$scommitid
-"
+%install
+rm -rf $RPM_BUILD_ROOT
+install -d $RPM_BUILD_ROOT/etc/minio/certs
+install -d $RPM_BUILD_ROOT/etc/systemd/system
+install -d $RPM_BUILD_ROOT/etc/default
+install -d $RPM_BUILD_ROOT/usr/local/bin
 
-%gobuild -o %{name} %{import_path}
+cat <<EOF >> $RPM_BUILD_ROOT/etc/default/minio
+# Remote volumes to be used for MinIO server.
+# Uncomment line before starting the server.
+# MINIO_VOLUMES=http://node{1...6}/export{1...32}
 
-# check that version set properly
-./%{name} version | tee v
+# Root credentials for the server.
+# Uncomment both lines before starting the server.
+# MINIO_ACCESS_KEY=Server-Access-Key
+# MINIO_SECRET_KEY=Server-Secret-Key
 
-#Version: 2016-09-11T17-42-18Z
-#Release-Tag: RELEASE.2016-09-11T17-42-18Z
-#Commit-ID: 85e2d886bcb005d49f3876d6849a2b5a55e03cd3
-v=$(awk '/Version:/{print $2}' v)
-test "$v" = $version
-v=$(awk '/Release-Tag:/{print $2}' v)
-test "$v" = $tag
-v=$(awk '/Commit-ID:/{print $2}' v)
-test "$v" = $commitid
+MINIO_OPTS="--certs-dir /etc/minio/certs"
+EOF
 
-%install
-rm -rf $RPM_BUILD_ROOT
-install -d $RPM_BUILD_ROOT%{_sbindir}
-install -p %{name} $RPM_BUILD_ROOT%{_sbindir}
+install %{_sourcedir}/minio.service $RPM_BUILD_ROOT/etc/systemd/system/minio.service
```

Comment:
I am not sure why you need to prefix here with `$RPM_BUILD_ROOT` and if you do need to, then the attr setting below would need the same path?

Reply:
this is because whatever is inside RPM_BUILD_ROOT gets packaged as RPM @kerneltime - its `%build` stage